synapse/handlers/profile.py

```diff
@@ -89,6 +89,14 @@ async def get_profile(self, user_id):
             except RequestSendFailed as e:
                 raise SynapseError(502, "Failed to fetch profile") from e
             except HttpResponseException as e:
+                if e.code < 500 and e.code != 404:
+                    # Other codes are not allowed in c2s API
+                    logger.error(
+                        "Server replied with wrong response: %s %s", str(e.code), e.msg
+                    )
+                    # Change to 500 not to confuse clients
+                    e.code = 500
+                    e.msg = "Remote server replied: {} {}".format(str(e.code), e.msg)
```

Comment:
No need to call str when passing to str.format. Same with the logging call I think.

Reply:
You are right, I fixed it.